src/full_space.jl

```diff
@@ -0,0 +1,224 @@
+"""
+    FullSpaceModel{T, S} <: AbstractNLPModel{T, S}   
+
+A data type to store the full space `BlockNLPModel` as an `AbstractNLPModel`.
+"""
+mutable struct FullSpaceModel{T, S} <: AbstractNLPModel{T, S}
+    meta::NLPModelMeta{T, S}
+    counters::Counters
+    blocknlp::AbstractBlockNLPModel
+  end
+
+"""
+FullSpaceModel(
+    ::Type{T}, m::AbstractBlockNLPModel
+    ) where {T}
+
+Converts a `BlockNLPModel` to a `AbstractNLPModel` that can be solved with any standard NLP solver.
+
+# Arguments
+
+- `m::AbstractBlockNLPModel`: name of the BlockNLPModel
+"""
+function FullSpaceModel(::Type{T}, m::AbstractBlockNLPModel) where {T}
```

Comment:
Why do we need `Type{T}` in the signature? Are we planning to work in multiprecision?

Reply:
We don't need it. I wrote a different version earlier where I needed to keep `Type{T}` there, but later changed it. I have removed it now. Thanks for pointing this out.